community/bolt/src/main/java/org/neo4j/bolt/transport/SocketTransportHandler.java

```diff
@@ -166,27 +167,31 @@ public ProtocolChooser( PrimitiveLongObjectMap<Function<Channel,BoltProtocol>> a
 
         public HandshakeOutcome handleVersionHandshakeChunk( ByteBuf buffer, Channel ch )
         {
-            if ( suggestedVersions.remaining() > buffer.readableBytes() )
+            if ( handShake.remaining() > buffer.readableBytes() )
             {
-                suggestedVersions.limit( suggestedVersions.position() + buffer.readableBytes() );
-                buffer.readBytes( suggestedVersions );
-                suggestedVersions.limit( suggestedVersions.capacity() );
+                handShake.limit( handShake.position() + buffer.readableBytes() );
+                buffer.readBytes( handShake );
+                handShake.limit( handShake.capacity() );
             }
             else
             {
-                buffer.readBytes( suggestedVersions );
+                buffer.readBytes( handShake );
             }
 
-            if ( suggestedVersions.remaining() == 0 )
+            if ( handShake.remaining() == 0 )
             {
-                suggestedVersions.flip();
-                for ( int i = 0; i < 4; i++ )
+                handShake.flip();
+                //Check so that handshake starts with 0x606 0B017
+                if ( handShake.getInt() == MAGIC_PREAMBLE )
```

Comment:
I would say it would be good if we have
```
if (handShake.getInt() == MAGIC)
{...}
else
{
  return HandshakeOutcome.WRONG_PROTOCOL;
}
```
and then in some outside method, we send back sth like "Error: your protocol is not recognized as bolt. Let's speak Chinese"